discussions-api/src/main/java/org/xwiki/contrib/discussions/DiscussionContextService.java

```diff
@@ -112,4 +112,16 @@ Optional<DiscussionContext> getOrCreate(String applicationHint, String name, Str
      * @return {@code true} if the current user can view the discussion context, {@code false} otherwise
      */
     boolean canViewDiscussionContext(DiscussionContextReference reference);
+
+    /**
+     * Save a new metadata to be added in the discussion context.
+     * This method both puts the metadata in {@link DiscussionContext#getMetadata()} map and also persists it
+     * permanently.
+     *
+     * @param context the context for which to add the new metadata.
+     * @param key the key of the metadata to save
+     * @param value the value of the metadata
+     * @return {@code true} if the metadata was properly saved.
+     */
+    boolean saveMetadata(DiscussionContext context, String key, String value);
```

Comment:
WDYT of adding a `boolean saveMetadata(DiscussionContext context, Map<String, String> metadata)` operation as well, to save several key/value pairs in a single operation?

Reply:
Well in that case I'd remove that one, to the one you propose. Might be better indeed on the long term.

Reply:
Well I don't think we need both tbh. If you need to only put a single value you can use a singleton map.